File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-2.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int slow = 0;
        int fast = 0;

        while (fast < nums.length) {
            if (nums[slow] == nums[fast]) {
                fast++;
                continue;
            } else {
                nums[slow + 1] = nums[fast];
                slow++;
            }
        }

        return slow + 1;
    }
}

File: Data Structures & Algorithms/find-duplicate-integer/submission-2.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        slow, fast = 0, 0
        while True:
            slow = nums[slow]
            fast = nums[nums[fast]]
            # If this is true, we are in the loop
            if fast == slow:
                break
        
        # Create a second slow pointer
        slowTwo = 0
        while True:
            slow = nums[slow]
            slowTwo = nums[slowTwo]
            if slow == slowTwo:
                break;
        
        return slow;